include/deal.II/lac/vector.h

```diff
@@ -191,12 +191,16 @@ class Vector : public Subscriptor
    * configuration time.
    *
    * Note that due to the communication model used in MPI, this operation can
-   * only succeed if all processes do it at the same time. This means that it
-   * is not possible for only one process to obtain a copy of a parallel
-   * vector while the other jobs do something else. This call will rather
-   * result in a copy of the vector on all processors.
-   */
-  explicit Vector (const TrilinosWrappers::MPI::Vector &v);
+   * typically only succeed if all processes do it at the same time. This means
+   * that it is not normally possible for only one or a subset of processes to
+   * obtain a copy of a parallel vector while the other jobs do something else.
+   * So, by default this call will result in a copy of the vector on all
+   * processors. If the @p use_local_mpi_communicator flag is set then the
+   * resulting local vector need only exist on the processes that have knowledge
+   * of @p v (i.e. those given by <code>v.get_mpi_communicator()</code>).
+   */
+  explicit Vector (const TrilinosWrappers::MPI::Vector &v,
+                   const bool                           use_local_mpi_communicator = false);
```

Comment:
Let’s see what others have to say, but an alternative is to take MPI_Comm = MPI_COMM_WORLD’ and make sure it contains communicator in the input vector.

Reply:
Personally, I think that @jppelteret fixed a bug and so I would not change the interface. If I use a copy constructor, I would expect to use the same communicator not MPI_COMM_WORLD.

Reply:
then maybe we should use the same communicator from `v.get_communicator()` and then not change the interface at all?

Reply:
Yes, that's what I think.